presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -396,7 +396,9 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
             return EMPTY_STATISTICS;
         }
         List<HivePartition> hivePartitions = partitionManager.getPartitions(metastore, tableHandle, constraint).getPartitions();
-        Map<String, ColumnHandle> tableColumns = getColumnHandles(session, tableHandle);
+        Map<String, ColumnHandle> tableColumns = Maps.filterValues(
```

Comment:
`filterValues` returns a view which can have unexpected performance characteristics. For this reason, we avoid using Guava methods that return collection views.

This could be wrapped with `ImmutableMap.copyOf()`, but we can easily use streams instead:
```java
Map<String, ColumnHandle> tableColumns = getColumnHandles(session, tableHandle)
        .entrySet().stream()
        .filter(entry -> ((HiveColumnHandle) entry.getValue()).isHidden())
        .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));